actors/builtin/miner/miner_state.go

```diff
@@ -939,11 +960,14 @@ func (st *State) AddInitialPledgeRequirement(amount abi.TokenAmount) {
 	st.InitialPledgeRequirement = newTotal
 }
 
-func (st *State) AddLockedFunds(store adt.Store, currEpoch abi.ChainEpoch, vestingSum abi.TokenAmount, spec *VestSpec) error {
+func (st *State) AddLockedFunds(vestingFunds *VestingFunds, currEpoch abi.ChainEpoch, vestingSum abi.TokenAmount, spec *VestSpec) error {
```

Comment:
This pattern (passing in the vesting funds object to the state like this) makes me uncomfortable.

Honestly, loading and storing vesting funds inside this function is probably not a huge issue and makes it much easier to reason about this code.

Reply:
@anorth This then ties to the issue of "batching" reads. What do we want to do here ?

Reply:
I was going to say the same thing as soon as I saw the first call to `UnlockVestedFunds`.

There is only one case where this pattern makes any difference, which is this one. In the others, the funds are loaded, mutated once, then stored. The load/store should be encapsulated in the state method.

At the call site of AddLockedFunds, first the vesting is processed and then a new amount locked up. We can just combine those into this call. Break the internals of `UnlockVestedFunds` out to a private method that can be called from here, and add a return value of the amount vested. Then encapsulate the load/store with no cost.